Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp

```diff
@@ -188,64 +188,74 @@ namespace AZ
         }
 
         Outcome<Data::Asset<MaterialAsset>> MaterialSourceData::CreateMaterialAssetFromSourceData(
-            Data::AssetId assetId, AZStd::string_view materialSourceFilePath, bool elevateWarnings, bool includeMaterialPropertyNames) const
+            Data::AssetId assetId,
+            AZStd::string_view materialSourceFilePath,
+            bool elevateWarnings,
+            bool includeMaterialPropertyNames,
+            AZStd::unordered_set<AZStd::string>* sourceDependencies) const
         {
-            MaterialAssetCreator materialAssetCreator;
-            materialAssetCreator.SetElevateWarnings(elevateWarnings);
+            const auto materialTypeSourcePath = AssetUtils::ResolvePathReference(materialSourceFilePath, m_materialType);
+            const auto materialTypeAssetId = AssetUtils::MakeAssetId(materialTypeSourcePath, 0);
+            if (!materialTypeAssetId.IsSuccess())
+            {
+                return Failure();
+            }
 
-#if 0
             MaterialTypeSourceData materialTypeSourceData;
-            AZStd::string materialTypeSourcePath = AssetUtils::ResolvePathReference(materialSourceFilePath, m_materialType);
             if (!AZ::RPI::JsonUtils::LoadObjectFromFile(materialTypeSourcePath, materialTypeSourceData))
             {
                 return Failure();
             }
 
             materialTypeSourceData.ResolveUvEnums();
 
-            auto materialTypeAsset =
-                materialTypeSourceData.CreateMaterialTypeAsset(AZ::Uuid::CreateRandom(), materialTypeSourcePath, elevateWarnings);
-            if (!materialTypeAsset.IsSuccess())
-            {
-                return Failure();
-            }
-#else
-            auto materialTypeAsset = AssetUtils::LoadAsset<MaterialTypeAsset>(materialSourceFilePath, m_materialType);
+            const auto materialTypeAsset =
+                materialTypeSourceData.CreateMaterialTypeAsset(materialTypeAssetId.GetValue(), materialTypeSourcePath, elevateWarnings);
             if (!materialTypeAsset.IsSuccess())
             {
                 return Failure();
             }
-#endif
 
-            materialAssetCreator.Begin(assetId, *materialTypeAsset.GetValue().Get(), includeMaterialPropertyNames);
+            AZStd::unordered_set<AZStd::string> dependencies;
+            dependencies.insert(materialSourceFilePath);
```

Comment:
Including materialSourceFilePath shouldn't be necessary. The material file shouldn't depend on itself. Or am I misunderstanding something?

Reply:
Should not depend on itself but needed to track circular references

Reply:
I don't see a problem with that. Could also rename the variable to something like sourceManifest or loadedAssets. Could also add the functor script files to it, anything we want to monitor for a reload.